usage_registry.go

```diff
@@ -26,29 +31,47 @@ type UsageRegistry interface {
 	StartReporting(eventSink Handler, reportInterval time.Duration, msgQueueSize int)
 }
 
-func NewUsageRegistry(sourceId string, tags map[string]string, closeNotify <-chan struct{}) UsageRegistry {
+type UsageRegistryConfig struct {
+	SourceId             string
+	Tags                 map[string]string
+	EventQueueSize       int
+	CloseNotify          <-chan struct{}
+	IntervalAgeThreshold time.Duration
+}
+
+func DefaultUsageRegistryConfig(sourceId string, closeNotify <-chan struct{}) UsageRegistryConfig {
+	return UsageRegistryConfig{
+		SourceId:             sourceId,
+		Tags:                 map[string]string{},
+		CloseNotify:          closeNotify,
+		EventQueueSize:       DefaultEventQueueSize,
+		IntervalAgeThreshold: DefaultIntervalAgeThreshold,
+	}
+}
+
+func NewUsageRegistry(config UsageRegistryConfig) UsageRegistry {
+	if config.EventQueueSize < 16 {
+		config.EventQueueSize = 16
+	}
+
+	if config.IntervalAgeThreshold < 1 {
```

Comment:
what is '1' here? 1 day? 1 second? seems like this should be compared to some duration no?

Reply:
Oh I was thinking it should be like 1 * time.second or minute or something. 1 nanosecond shouldn't be valid should it?

Reply:
Because this was a structural cleanup to move a global into the per-registry state, I'm just maintaining existing behavior. I could have if < min, assign min, but since in most cases we don't set the value we'd be assigned the min value instead of the default value.